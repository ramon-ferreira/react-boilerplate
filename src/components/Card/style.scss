@font-face {
  font-family: 'Raleway';
  src: url('@assets/fonts/Raleway-VariableFont_wght.ttf') format('truetype');
}

.card {
  display: flex;
  flex-direction: column;
  padding: 1rem;
  font-family: Raleway;
  font-size: 2rem;
}

.border {
  border-style: solid;
  border-width: 2px;
  border-radius: 15px;
}
