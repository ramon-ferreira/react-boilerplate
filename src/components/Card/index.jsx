import { PropTypes } from 'prop-types';

import style from './style.scss';

function Card ({ border = true, children }) {
  const classes = [style.card];

  border && classes.push(style.border);

  return (
    <div className={classes.join(' ')}>
      {children}
    </div>
  );
}

Card.propTypes = {
  border: PropTypes.bool,
  children: PropTypes.node
};

export default Card;
