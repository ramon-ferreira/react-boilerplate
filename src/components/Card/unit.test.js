import renderer from 'react-test-renderer';
import Card from './index.jsx';

test('Component render', () => {
  const component = renderer.create(
    <Card></Card>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/** Unit test example */
