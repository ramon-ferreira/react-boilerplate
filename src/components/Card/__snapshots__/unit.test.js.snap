// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render 1`] = `
<div
  className="card border"
/>
`;
